packages/client/screens/filters/FilterScreen.tsx

```diff
@@ -132,9 +146,41 @@ export default function FilterScreen({ route, navigation }: HomeStackScreenProps
             <RadioGroup options={filtersByDate} value={filterByDate} setValue={setFilterByDate} />
             <Box marginTop="20px" />
             {filterByDate === filtersByDate[filtersByDate.length - 1] && <CustomDates />}
-            {!(filterByTransaction === filtersByTransaction[0] && filterByDate === filtersByDate[0]) &&
+            <Pressable
+                marginTop="30px"
+                onPress={() =>
+                    navigation.navigate("FilterContactScreen", {
+                        filters: route.params.filters,
+                        setFilters: route.params.setFilters,
+                        contactFilters: route.params.contactFilters,
+                        setContactFilters: route.params.setContactFilters,
+                    })
+                }
+                _pressed={{
+                    background: "text.200",
+                }}
+            >
+                <HStack height="50px" alignItems="center">
+                    <Text fontWeight={"semibold"} color={"text.500"} marginRight="15">
+                        Contacts
+                    </Text>
+                    {filterByContact.map((contact) => (
+                        <Text color={"text.500"} marginRight="1">
+                            {contact},
+                        </Text>
+                    ))}
+                    <FontAwesomeIcon icon={farChevronRight} style={styles.chevronRightIcon} size={16} />
+                </HStack>
+            </Pressable>
+            <Box marginTop="20px" />
+            {!(
+                filterByTransaction === filtersByTransaction[0] &&
+                filterByDate === filtersByDate[0] &&
+                filterByContact.length === 0
+            ) &&
                 (!isFilterSaved ||
-                    !(filterByTransaction === route.params.filters[0] && filterByDate === route.params.filters[1])) && (
+                    (!(filterByTransaction === route.params.filters[0] && filterByDate === route.params.filters[1]) &&
+                        filterByContact.length === 0)) && (
```

Comment:
I think it should be like this no? where its inside the negation. I might be wrong though if you tested all the different cases, the logic gets confusing
```suggestion
                    (!(filterByTransaction === route.params.filters[0] && filterByDate === route.params.filters[1] && filterByContact.length === 0))) && (
```